cpp/tests/CMakeLists.txt

```diff
@@ -199,14 +199,17 @@ if(BUILD_TESTS)
   )
 
   if(BUILD_CAGRA_HNSWLIB)
-    ConfigureTest(NAME NEIGHBORS_HNSW_TEST PATH neighbors/hnsw.cu GPUS 1 PERCENT 100)
+    ConfigureTest(
+      NAME NEIGHBORS_HNSW_TEST PATH neighbors/hnsw.cu GPUS 1 PERCENT 100 ADDITIONAL_DEP NCCL::NCCL
```

Comment:
Why was the link needed here? AFAIK this code path does not use NCCL.

Reply:
Do you have the trace? This seems like a leak to me. No problem if not, I can track it down - but shouldn't hold this PR on that, can fix as a follow up.

Reply:
re-testing locally and I can't reproduce the nccl dependency. So I am dropping this change